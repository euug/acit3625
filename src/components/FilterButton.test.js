import React from 'react';
import renderer from 'react-test-renderer';
import FilterButton from './FilterButton';
import { mount } from 'enzyme';
import "../setupTests";

describe('Render Tests', () => {
    test('Snapshot Test', () => {
        const component = renderer.create(<FilterButton />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});

describe('Button Event Test', () => {
    test('FilterButton Event Test', () => {
        const setFilter = jest.fn();
        const wrapper = mount(<FilterButton setFilter={setFilter}>Button</FilterButton>);
        const filter = wrapper.find('button');
    
        filter.simulate('click');
        expect(setFilter).toHaveBeenCalled();
    });
});