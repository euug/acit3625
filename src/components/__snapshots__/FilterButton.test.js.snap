// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton Component Tests Renders Correctly 1`] = `
<button
  className="btn toggle-btn"
  onClick={[Function]}
  type="button"
>
  <span
    className="visually-hidden"
  >
    Show 
  </span>
  <span />
  <span
    className="visually-hidden"
  >
     tasks
  </span>
</button>
`;

exports[`Render Tests Snapshot Test 1`] = `
<button
  className="btn toggle-btn"
  onClick={[Function]}
  type="button"
>
  <span
    className="visually-hidden"
  >
    Show 
  </span>
  <span />
  <span
    className="visually-hidden"
  >
     tasks
  </span>
</button>
`;

exports[`Snapshot Test Renders Correctly 1`] = `
<button
  className="btn toggle-btn"
  onClick={[Function]}
  type="button"
>
  <span
    className="visually-hidden"
  >
    Show 
  </span>
  <span />
  <span
    className="visually-hidden"
  >
     tasks
  </span>
</button>
`;
